### ISSUE TITLE
warning when compiling on musl libc

### ISSUE THREAD

**Issue by rofl0r** [chriskohlhoff/asio#204]
this pops up in all compilation units:
```
In file included from ./../../../include/asio/detail/socket_types.hpp:61:0,
                 from ./../../../include/asio/impl/error_code.ipp:29,
                 from ./../../../include/asio/error_code.hpp:185,
                 from ./../../../include/asio/detail/scheduler.hpp:20,
                 from ./../../../include/asio/system_executor.hpp:19,
                 from ./../../../include/asio/associated_executor.hpp:21,
                 from ./../../../include/asio.hpp:19,
                 from iostreams/daytime_client.cpp:13:
/usr/include/sys/poll.h:1:2: warning: #warning redirecting incorrect #include <sys/poll.h> to <poll.h> [-Wcpp]
 #warning redirecting incorrect #include <sys/poll.h> to <poll.h>
  ^~~~~~~
```
the warning is because sys/poll.h is wrong, it should be just poll.h, as mentioned in POSIX (`man 3p poll`).

**Comment by ryanfb** [comment]
This also causes MAME compiles to halt, because the set `-Werror` by default.

**Comment by chriskohlhoff** [comment]
I applied a fixed in 5adce6e26769228f2c9fcc0b9dda8bf00279821f. As older UNIXes (non linux, non-bsd) may still require `sys/poll.h` I conservatively enabled the change only for those I can confirm have `poll.h`.

**Comment by rofl0r** [comment]
the fix should be the other way round: use poll.h by default and enable sys/poll.h only on known bad systems. http://wiki.musl-libc.org/wiki/FAQ#Q:_application_XY_misbehaves_or_crashes_at_runtime_when_linked_against_musl
